gateway-service/src/main/java/org/zowe/apiml/gateway/conformance/ValidateAPIController.java

```diff
@@ -10,96 +10,164 @@
         produces = MediaType.APPLICATION_JSON_VALUE
     )
-    public ResponseEntity<Object> checkValidate(@RequestBody String serviceID) {
+    @HystrixCommand
+    public ResponseEntity<String> checkConformance(@PathVariable String serviceId) {
+
+
+        ConformanceProblemsContainer foundNonConformanceIssues = new ConformanceProblemsContainer();
 
-        String message = validator(serviceID);
 
-        if (!message.isEmpty()) {
-            return new ResponseEntity<>(messageService.createMessage(invalidKey, message).mapToApiMessage(), HttpStatus.BAD_REQUEST);
+        foundNonConformanceIssues.put("Registration problems", validatorItem1(serviceId));
+
+        if (foundNonConformanceIssues.size() != 0) {     // cant continue if a service isn't registered
+            return GenerateBadRequestResponseEntity(wrongServiceIdKey, foundNonConformanceIssues);
         }
 
-        return new ResponseEntity<>(HttpStatus.OK);
+        foundNonConformanceIssues.put("Metadata problems", metaDataCheck(serviceId));
+
+        if (foundNonConformanceIssues.size() != 0) {     // cant continue without metadata
+            return GenerateBadRequestResponseEntity(NoMetadataKey, foundNonConformanceIssues);
+        }
 
+        foundNonConformanceIssues.put("Conformance problems", validatorItem5(serviceId));
+
+        if (foundNonConformanceIssues.size() != 0) {
+            return GenerateBadRequestResponseEntity(NonConformantKey, foundNonConformanceIssues);
+        }
+
+//        return new ResponseEntity<>(HttpStatus.OK);
+
+        return new ResponseEntity<>("{\"message\":\"Service fulfills all checked conformance criteria\"}"  ,HttpStatus.OK);
     }
 
+
     /**
-     * Accept serviceID and check if it contains only lower case characters without symbols, return true if it meets the criteria
-     * otherwise return false
+     * Mapping so the old endpoint keeps working.
      *
-     * @param serviceID accept serviceID to check
-     * @return return boolean variable False if it only contains lower case characters without symbols
+     * @param serviceId serviceId to check for conformance
+     * @return 200 if service is conformant, 400 + JSON explanation if not
      */
-    private boolean checkValidPatternAPI(String serviceID) {
+    @PostMapping(value = "/validate", produces = MediaType.APPLICATION_JSON_VALUE)
+    @HystrixCommand
+    public ResponseEntity<String> checkValidateLegacy(@RequestBody String serviceId) {
+        return checkConformance(serviceId);
+    }
 
-        Matcher findSymbol = symbolPattern.matcher(serviceID);
 
-        return findSymbol.find();
+    /**
+     * Creates a response when a conformance criteria is failed.
+     *
+     * @param foundNonConformanceIssues list of found issues
+     * @return Response that this controller returns
+     */
+    private ResponseEntity<String> GenerateBadRequestResponseEntity(String key, ConformanceProblemsContainer foundNonConformanceIssues) {
```

Comment:
change the name GenerateBadRequestResponseEntity() to generateBadRequestResponseEntity()